qiskit/aqua/operators/gradients/circuit_gradients/param_shift.py

```diff
@@ -261,9 +266,16 @@ def get_primitives(item):
                 item = item.primitive.data
             return item
 
+        is_statefn = False
         if isinstance(x, Iterable):
```

Comment:
Looking at the typehint above, this is the case of `x` being a list? I think it would be a little bit clearer if we could change this to be the same as above 😄 
```python
if isinstance(x, list):
```

Reply:
Well in principle I suppose we could drop the 'StateFn' check. In my point of view the type hint supports the user to get more insight on what is going on.